db-service/test/cqn4sql/calculated-elements.test.js

```diff
@@ -98,17 +169,116 @@ describe.skip('Unfolding calculated elements in select list', () => {
+    `
+    expect(query).to.deep.equal(expected)
+  })
+
+  it('in group by & having', () => {
+    let query = cqn4sql(
+      CQL`SELECT from booksCalc.Books { ID, sum(price) as tprice }
+      group by ctitle having ctitle like 'A%'`,
+      model,
+    )
+    const expected = CQL`SELECT from booksCalc.Books as Books {
+        Books.ID, sum(Books.price) as tprice
+      } group by substring(Books.title, 3, Books.stock)
+        having substring(Books.title, 3, Books.stock) like 'A%'
+    `
+    expect(query).to.deep.equal(expected)
+  })
+
+  it('in order by', () => {
+    let query = cqn4sql(CQL`SELECT from booksCalc.Books { ID, title } order by ctitle`, model)
+    const expected = CQL`SELECT from booksCalc.Books as Books {
+        Books.ID, Books.title
+      } order by substring(Books.title, 3, Books.stock)
+    `
+    expect(query).to.deep.equal(expected)
+  })
+
+  it('in filter in path in FROM', () => {
+    let query = cqn4sql(CQL`SELECT from booksCalc.Authors[name like 'A%'].books[storageVolume < 4] { ID }`, model)
+    const expected = CQL`SELECT from booksCalc.Books as books {
+      books.ID
+    } where exists (select 1 from booksCalc.Authors as Authors 
+                      where Authors.ID = books.author_ID
+                        and (Authors.firstName || ' ' || Authors.lastName) like 'A%')
+                        and (books.stock * ((books.length * books.width) * books.height)) < 4
+    `
+    expect(query).to.deep.equal(expected)
+  })
+})
+
+// ? calc elem at several places in one query (select, where, order ...) ?
+
+describe('Unfolding calculated elements ... misc', () => {
+  let model
+  beforeAll(async () => {
+    model = cds.model = await cds.load(__dirname + '/model/booksWithExpr').then(cds.linked)
+  })
+  // Calculated elements on-write are not supported, yet (in entity:“booksCalc.Books”/element:“areaS”)
+  it.skip('calculated element on-write (stored) is not unfolded', () => {
+    let query = cqn4sql(CQL`SELECT from booksCalc.Books { ID, areaS }`, model)
+    const expected = CQL`SELECT from booksCalc.Books as Books { Books.ID, Books.areaS }`
+    expect(query).to.deep.equal(expected)
+  })
+})
+
+describe('Unfolding calculated elements and localized', () => {
+  let model
+  beforeAll(async () => {
+    model = cds.model = await cds.load(__dirname + '/model/booksWithExpr').then(cds.linked)
+    model = cds.compile.for.nodejs(model)
+  })
+
+  it('presence of localized element should not affect unfolding', () => {
+    const q = CQL`SELECT from booksCalc.LBooks { ID, title, area }`
+    q.SELECT.localized = true
+    let query = cqn4sql(q, model)
+    const expected = CQL`SELECT from localized.booksCalc.LBooks as LBooks {
+        LBooks.ID,
+        LBooks.title,
+        LBooks.length * LBooks.width as area
+      }`
+    expected.SELECT.localized = true
+    expect(query).to.deep.equal(expected)
+  })
+
+  it.skip('calculated element refers to localized element', () => {
```

Comment:
@stewsk FYI: had to skip this and comment out the calc element which points to a localized element. Will re-enable once compiler 4.1 is on npm